### PR TITLE
Fix crash on invalid part size

### PR DESCRIPTION
**Pull request by passaro**
## Description of change

The CRT changed the allowed values for `part_size` in a recent update (merged in #633). Because of that change, launching Mountpoint with previously accepted part sizes may result in a crash, e.g.:

```
> mount-s3 <BUCKET> <DIR> --foreground --part-size 10000000 
[...] panic on Thread { id: ThreadId(1), name: Some("main"), .. } at mountpoint-s3-client/src/s3_crt_client.rs:304:64: called `Result::unwrap()` on an `Err` value: Error(34, "aws-c-common: AWS_ERROR_INVALID_ARGUMENT, An invalid argument was passed to a function.")
```

This change prevents the panic and instead reports an error. Note that the error message is currently not too informative. We plan to improve it in a follow up change.

Example:
```
> mount-s3 <BUCKET> <DIR> --foreground --part-size 10000000 --debug-crt
[...] ERROR awscrt::S3Client: Failed to initialize buffer pool. Chunk size must be either 0 or more than 1 KB and size must be 1 KB aligned.    
Caused by:
    0: Unknown CRT error
    1: CRT error 34: aws-c-common: AWS_ERROR_INVALID_ARGUMENT, An invalid argument was passed to a function.
```

## Does this change impact existing behavior?

N/A

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

File: mountpoint-s3-client/src/s3_crt_client.rs [hunk 1]
```diff
@@ -769,10 +769,24 @@ pub enum NewClientError {
...
     InvalidConfiguration(String),
+    /// An internal error from within the AWS Common Runtime
+    #[error("Unknown CRT error")]
+    CrtError(#[source] mountpoint_s3_crt::common::error::Error),
+}
+
+impl NewClientError {
+    fn construction_error(e: mountpoint_s3_crt::common::error::Error) -> Self {
+        const AWS_ERROR_INVALID_ARGUMENT: i32 = 34;
+        if e.raw_error() == AWS_ERROR_INVALID_ARGUMENT {
```
Comment (monthonk) [inline mountpoint-s3-client/src/s3_crt_client.rs:784]: I'm a bit uncomfortable about using hard-coded number. Should we add a function `is_invalid_argument_error` in `mountpoint_s3_crt::common::error::Error` and use the enum from Rust binding directly?

File: mountpoint-s3-client/src/s3_crt_client.rs [hunk 2]
```diff
@@ -769,10 +769,24 @@ pub enum NewClientError {
...
+            Self::InvalidConfiguration("see crt logs for details".into())
```
Comment (monthonk) [inline mountpoint-s3-client/src/s3_crt_client.rs:785]: Would using `e.to_string()` gives a bit better hint for the error?

**Review by monthonk — COMMENTED** [review]
Thanks for fixing this @passaro , added a few comments.

File: mountpoint-s3-client/src/s3_crt_client.rs (hunk 2 above)
Comment (passaro) [reply mountpoint-s3-client/src/s3_crt_client.rs]: I've removed this method and just wrapped the CRT error for now. The error message may be even less informative, but we can fix that at a later time.

**Review by monthonk — APPROVED** [review]
LGTM!

**Pull request merged**